### ISSUE TITLE
Support for "long" number format

### ISSUE THREAD

**Issue by DannyHinshaw** [deepmap/oapi-codegen#421]
Hello! First of all thank you very much for this project, this is great!

I am working on using this tool to generate a client and types from an OAPI v3 spec that contains some number types with "long" format like this:

```json
{
          "someNumberField" : {
            "type" : "number",
            "description" : "Some description.",
            "format" : "long",
            "example" : 100
          }
}
```

Currently this is throwing an error when I attempt to do generation:

```
error resolving primitive type: invalid number format: long
```

This appears to be a valid OAPI v3 number format, at least it passes [validation here](https://github.com/deepmap/oapi-codegen). So I was wondering if you would be open to adding it to the [supported formats here](https://github.com/deepmap/oapi-codegen/blob/1608d14623301ada321842225ee06489b627e141/pkg/codegen/schema.go#L371).

I'd be more than happy to submit a PR for this if so!

**Comment by Taillook** [comment]
"long" is int64?

If int64, then like this:
```yaml
someNumberField:
  type: integer
  description: Some description.
  format: int64
  example: 100

```

ref: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#data-types

**Comment by DannyHinshaw** [comment]
@Taillook yes, I would like it to convert to `int64`, I do not maintain the OAPI specs though, so I'm hoping to add support to the tool.

**Comment by Taillook** [comment]
I see, understood.(just doubted.

**Comment by mromaszewicz** [comment]
The original error 'invalid number format: long' no longer exists in the codebase. PR #2223 replaced hardcoded format switch statements with a configurable TypeMapping system whose Resolve() method falls back gracefully to the type default for unknown formats. Running codegen against a spec with number/long now succeeds (producing float32), and users who want int64 for 'long' can configure output-options.type-mapping.number.formats.long=int64.

Closing as fixed. If you can still reproduce on a recent release, please reopen with the exact spec and version used.
